src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseGradingService.java

```diff
@@ -318,6 +369,35 @@ private void updateScore(Result result, Set<ProgrammingExerciseTestCase> success
                 result.getFeedbacks().stream().filter(fb -> fb.getText().equals(test.getTestName())).findFirst().ifPresent(feedback -> feedback.setCredits(testPointsWithBonus));
                 return testPointsWithBonus;
             }).sum();
+
+            if (programmingExercise.isStaticCodeAnalysisEnabled() && Optional.ofNullable(programmingExercise.getMaxStaticCodeAnalysisPenalty()).orElse(0) > 0) {
```

Comment:
please add some inline comments what you are doing here! Otherwise the code is hard to understand!

Reply:
I would also suggest to extract two additional methods here.
One takes case about the tests, the other one about static code analysis.
The method `updateScore` takes care about invoking both (the 2nd only if static code analysis is enabled)

Reply:
I improved the readability and separated the penalty calculation into a new function.